### ISSUE TITLE
F-Droid

### ISSUE THREAD

**Issue by dimqua** [fython/PackageTracker#67]
I think it'd be nice to add such a good app to the F-Droid app store. :-)
Don't you mind? (I have to ask.)

Currently the app uses Firebase SDK, Google Play Services and a few [JARs](https://github.com/fython/PackageTracker/tree/3.0.3-49/mobile/libs). They are prohibited by F-Droid rules. Is it possible to build this app without them?

Besides, the latest Play Store release (3.1.4) is not tagged. Is this a mistake?

**Comment by IzzySoft** [comment]
Also, our scanner found some JARs (we don't allow JARs, all must be build from source):

    WARNING: Found JAR file at mobile/libs/ZHConverter.jar
    WARNING: Found JAR file at mobile/libs/pinyin4j-2.5.0.jar

Could you please take care for them as well? You could e.g. include their source trees as git submodules.

**Comment by fython** [comment]
@dimqua It's difficult to remove FCM because we use it to receive push message for latest status. I will consider a way to balance the two and try to add into the F-Droid store.

**Comment by fython** [comment]
@IzzySoft Oops... Thanks for your reminder. I should obey license carefully. These days I will check it.

**Comment by IzzySoft** [comment]
Thanks @fython! As for FCM: there are alternatives. I don't know them in detail (I'm no dev), but e.g. Tutanota has implemented their own, plus there are 2 (or more?) FOSS libraries I've heard from (unfortunately I always forget their names).

**Comment by fython** [comment]
@IzzySoft How does Tutanota implement instant push service, use a foreground service and keep connection to their own servers (or user custom servers)? I want to try not to do this. It's not good if every apps keep their own push service. They will cost a lot of battery life.

In China Mainland, Google & API have been blocked for many years. Most apps (including not only social apps) build own service, which causes my phone runs slowly. That's a bad experience.

I expect a solution to provide a push service similar to FCM: A push library for our apps and a backend app for keeping a connection and forwarding push message to each apps. If anyone find some solutions, please share with me. I will be very grateful! Thanks!

**Comment by IzzySoft** [comment]
> How does Tutanota implement instant push service …

I'm no dev, so I cannot explain – but there was an Article on it at F-Droid: [How Tutanota replaced Google’s FCM with their own notification system](https://f-droid.org/en/2018/09/03/replacing-gcm-in-tutanota.html). And I wish I'd remember the other two…

**Comment by IzzySoft** [comment]
@fython any progress on this? My [snippet here](https://gitlab.com/snippets/1896503) contains a few hints on push messaging alternatives which might help. Do you see a chance to get the app "F-Droid ready"?

**Comment by fython** [comment]
@IzzySoft Thank you for still paying attention to the progress of my project. Since the API used by the project is temporarily unavailable, I put aside its update until I have a more suitable API to use, and the preparations for release on F-Droid may become more uncertain.

I am sorry to bring this news. At the same time, thank you very much for sharing this information. I will consider trying free push service solutions in other projects.

**Comment by IzzySoft** [comment]
Thanks for the update! Indeed I was hoping for better news – but can't be helped… Best luck then! :crossed_fingers: Just let us know when we can pick up again.

**Comment by IzzySoft** [comment]
@fython looks like my wishes have not been granted: didn't hear back from you, and there were no changes to the code published either. I hope it's OK with you if we close the [request at F-Droid's end](https://gitlab.com/fdroid/rfp/-/issues/884), as it's unlikely we can process it. Should the situation change, be welcome to ping us and we can pick up again.

**Comment by fython** [comment]
> @fython looks like my wishes have not been granted: didn't hear back from you, and there were no changes to the code published either. I hope it's OK with you if we close the [request at F-Droid's end](https://gitlab.com/fdroid/rfp/-/issues/884), as it's unlikely we can process it. Should the situation change, be welcome to ping us and we can pick up again.

@IzzySoft No problem. I'm sorry for not pushing forward because I don’t have enough time to maintain the project. Anyway, thank you very much for your help and follow up.

**Comment by fython** [comment]
Just keep this issue open because it's still a good feature to support.

**Comment by IzzySoft** [comment]
Yes, please do! I just had to apply some house-keeping on our end as it began to become unmaintainable. Hundreds of open issues with many we just could process, which were stalled, etc.
